esp-wifi/src/lib.rs

```diff
@@ -287,7 +288,7 @@ impl<'d> Drop for EspWifiController<'d> {
 ///
 /// This trait is meant to be used only for the `init` function.
 /// Calling `timers()` multiple times may panic.
-pub trait EspWifiTimerSource {
+pub trait EspWifiTimerSource: private::Sealed {
```

Comment:
Isn't this a bit of an overkill?

Reply:
Maybe, but is there a down side to doing it?

Reply:
More code :)

Reply:
I suppose, but I can't see a reason to keep this trait open of downstream crates to impl.

I'm happy to remove it, but I'd like a more compelling argument than removing a few lines of hidden boiler plate impls.